xio-core/src/main/java/com/xjeffrose/xio/http/ProxyClientFactory.java

```diff
@@ -29,7 +29,10 @@ public Client createClient(ChannelHandlerContext ctx, ClientConfig config) {
 
   @Override
   public Client getClient(ChannelHandlerContext ctx, ClientConfig config) {
-    return getHandlerClient(ctx)
-        .orElse(clientPool.acquire(config, () -> super.getClient(ctx, config)));
+    Client client =
+        getHandlerClient(ctx)
+            .orElse(clientPool.acquire(config, () -> super.getClient(ctx, config)));
```

Comment:
you could call `createClient` here rather than `super.getClient`  since we're always updating the channel attribute now